src/dev/performance/navigation.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import _ from 'lodash';
+import puppeteer from 'puppeteer';
+import { ToolingLog } from '@kbn/dev-utils';
+import { LoadingFinishedEvent, ResponseReceivedEvent } from './event';
+
+export async function navigateToApp(
+  log: ToolingLog,
+  options: { headless: boolean; appConfig: { url: string; login: string; password: string } }
+) {
+  const { headless, appConfig } = options;
+  const browser = await puppeteer.launch({ headless });
+  const devToolsResponses = new Map<string, any>();
+  const apps = [
+    'kibana',
```

Comment:
I believe the artifact size should be the same on every page. We can land on any page and reduce the test execution time.

Reply:
I think we only need to load a couple apps, not all of them, but the goal is to determine which assets are always loaded regardless of the application. Maybe we need more than three or four apps to do that correctly. We could also load the apps in parallel tabs if we wanted to speed things up.

Reply:
I did some testing and if I leave 

```
  const apps = [
    'kibana',
    'canvas', // +1 asset
    'maps', // +16
    'timelion', // +1
    'apm', // +1
    'uptime', // +1
  ]
```

we got 130 assets:

```
http://localhost:5620/bundles/app/core/bootstrap.js : 2655 bytes
http://localhost:5620/9007199254740991/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.@elastic.js : 1510806 bytes
http://localhost:5620/9007199254740991/bundles/plugin/kibanaUtils/kibanaUtils.plugin.js : 237821 bytes
http://localhost:5620/9007199254740991/bundles/plugin/kibanaReact/kibanaReact.plugin.js : 289148 bytes
http://localhost:5620/9007199254740991/bundles/plugin/esUiShared/esUiShared.plugin.js : 270365 bytes
http://localhost:5620/9007199254740991/bundles/plugin/data/data.plugin.js : 582819 bytes
http://localhost:5620/9007199254740991/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js : 2818962 bytes
http://localhost:5620/9007199254740991/bundles/plugin/usageCollection/usageCollection.plugin.js : 8979 bytes
http://localhost:5620/9007199254740991/bundles/plugin/licensing/licensing.plugin.js : 5560 bytes
http://localhost:5620/9007199254740991/bundles/plugin/observability/observability.plugin.js : 40054 bytes
http://localhost:5620/9007199254740991/bundles/plugin/telemetry/telemetry.plugin.js : 20189 bytes
http://localhost:5620/9007199254740991/bundles/plugin/testbed/testbed.plugin.js : 2024 bytes
http://localhost:5620/9007199254740991/bundles/plugin/newsfeed/newsfeed.plugin.js : 7143 bytes
http://localhost:5620/9007199254740991/bundles/plugin/mapsLegacyLicensing/mapsLegacyLicensing.plugin.js : 1858 bytes
http://localhost:5620/9007199254740991/bundles/plugin/kibanaLegacy/kibanaLegacy.plugin.js : 310502 bytes
http://localhost:5620/9007199254740991/bundles/plugin/devTools/devTools.plugin.js : 285399 bytes
http://localhost:5620/9007199254740991/bundles/plugin/uiActions/uiActions.plugin.js : 87679 bytes
http://localhost:5620/9007199254740991/bundles/plugin/mapsLegacy/mapsLegacy.plugin.js : 704205 bytes
http://localhost:5620/9007199254740991/bundles/plugin/statusPage/statusPage.plugin.js : 1947 bytes
http://localhost:5620/9007199254740991/bundles/plugin/share/share.plugin.js : 26812 bytes
http://localhost:5620/9007199254740991/bundles/plugin/inspector/inspector.plugin.js : 108907 bytes
http://localhost:5620/9007199254740991/bundles/plugin/embeddable/embeddable.plugin.js : 156068 bytes
http://localhost:5620/9007199254740991/bundles/plugin/advancedUiActions/advancedUiActions.plugin.js : 175343 bytes
http://localhost:5620/9007199254740991/bundles/plugin/embeddableEnhanced/embeddableEnhanced.plugin.js : 177690 bytes
http://localhost:5620/9007199254740991/bundles/plugin/drilldowns/drilldowns.plugin.js : 182870 bytes
http://localhost:5620/9007199254740991/bundles/plugin/bfetch/bfetch.plugin.js : 6416 bytes
http://localhost:5620/9007199254740991/bundles/plugin/cloud/cloud.plugin.js : 2033 bytes
http://localhost:5620/9007199254740991/bundles/plugin/expressions/expressions.plugin.js : 218383 bytes
http://localhost:5620/9007199254740991/bundles/plugin/home/home.plugin.js : 286595 bytes
http://localhost:5620/9007199254740991/bundles/plugin/charts/charts.plugin.js : 282286 bytes
http://localhost:5620/9007199254740991/bundles/plugin/apmOss/apmOss.plugin.js : 2203 bytes
http://localhost:5620/9007199254740991/bundles/plugin/console/console.plugin.js : 289288 bytes
http://localhost:5620/9007199254740991/bundles/plugin/searchprofiler/searchprofiler.plugin.js : 291217 bytes
http://localhost:5620/9007199254740991/bundles/plugin/painlessLab/painlessLab.plugin.js : 329566 bytes
http://localhost:5620/9007199254740991/bundles/plugin/grokdebugger/grokdebugger.plugin.js : 287719 bytes
http://localhost:5620/9007199254740991/bundles/plugin/management/management.plugin.js : 298184 bytes
http://localhost:5620/9007199254740991/bundles/plugin/upgradeAssistant/upgradeAssistant.plugin.js : 315302 bytes
http://localhost:5620/9007199254740991/bundles/plugin/reporting/reporting.plugin.js : 447057 bytes
http://localhost:5620/9007199254740991/bundles/plugin/licenseManagement/licenseManagement.plugin.js : 300876 bytes
http://localhost:5620/9007199254740991/bundles/plugin/indexManagement/indexManagement.plugin.js : 528166 bytes
http://localhost:5620/9007199254740991/bundles/plugin/indexLifecycleManagement/indexLifecycleManagement.plugin.js : 309466 bytes
http://localhost:5620/9007199254740991/bundles/plugin/crossClusterReplication/crossClusterReplication.plugin.js : 309711 bytes
http://localhost:5620/9007199254740991/bundles/plugin/remoteClusters/remoteClusters.plugin.js : 307006 bytes
http://localhost:5620/9007199254740991/bundles/plugin/watcher/watcher.plugin.js : 304238 bytes
http://localhost:5620/9007199254740991/bundles/plugin/indexPatternManagement/indexPatternManagement.plugin.js : 312199 bytes
http://localhost:5620/9007199254740991/bundles/plugin/advancedSettings/advancedSettings.plugin.js : 504628 bytes
http://localhost:5620/9007199254740991/bundles/plugin/telemetryManagementSection/telemetryManagementSection.plugin.js : 508657 bytes
http://localhost:5620/9007199254740991/bundles/plugin/visTypeVislib/visTypeVislib.plugin.js : 480456 bytes
http://localhost:5620/9007199254740991/bundles/plugin/fileUpload/fileUpload.plugin.js : 231008 bytes
http://localhost:5620/9007199254740991/bundles/plugin/dataEnhanced/dataEnhanced.plugin.js : 159238 bytes
http://localhost:5620/9007199254740991/bundles/plugin/visTypeVega/visTypeVega.plugin.js : 738385 bytes
http://localhost:5620/9007199254740991/bundles/plugin/visualizations/visualizations.plugin.js : 452955 bytes
http://localhost:5620/9007199254740991/bundles/plugin/features/features.plugin.js : 2805 bytes
http://localhost:5620/9007199254740991/bundles/plugin/visTypeTimeseries/visTypeTimeseries.plugin.js : 106127 bytes
http://localhost:5620/9007199254740991/bundles/plugin/visTypeTimelion/visTypeTimelion.plugin.js : 246581 bytes
http://localhost:5620/9007199254740991/bundles/plugin/rollup/rollup.plugin.js : 334761 bytes
http://localhost:5620/9007199254740991/bundles/plugin/snapshotRestore/snapshotRestore.plugin.js : 306896 bytes
http://localhost:5620/9007199254740991/bundles/plugin/security/security.plugin.js : 323351 bytes
http://localhost:5620/9007199254740991/bundles/plugin/beats_management/beats_management.plugin.js : 403088 bytes
http://localhost:5620/9007199254740991/bundles/plugin/transform/transform.plugin.js : 300645 bytes
http://localhost:5620/9007199254740991/bundles/plugin/canvas/canvas.plugin.js : 1725829 bytes
http://localhost:5620/9007199254740991/bundles/plugin/visTypeMarkdown/visTypeMarkdown.plugin.js : 186648 bytes
http://localhost:5620/9007199254740991/bundles/plugin/visTypeTagcloud/visTypeTagcloud.plugin.js : 301170 bytes
http://localhost:5620/9007199254740991/bundles/plugin/visTypeTable/visTypeTable.plugin.js : 434575 bytes
http://localhost:5620/9007199254740991/bundles/plugin/ingestPipelines/ingestPipelines.plugin.js : 517124 bytes
http://localhost:5620/9007199254740991/bundles/plugin/inputControlVis/inputControlVis.plugin.js : 98803 bytes
http://localhost:5620/9007199254740991/bundles/plugin/savedObjects/savedObjects.plugin.js : 104579 bytes
http://localhost:5620/9007199254740991/bundles/plugin/visTypeMetric/visTypeMetric.plugin.js : 297289 bytes
http://localhost:5620/9007199254740991/bundles/plugin/tileMap/tileMap.plugin.js : 725399 bytes
http://localhost:5620/9007199254740991/bundles/plugin/regionMap/regionMap.plugin.js : 717822 bytes
http://localhost:5620/9007199254740991/bundles/plugin/navigation/navigation.plugin.js : 83801 bytes
http://localhost:5620/9007199254740991/bundles/plugin/maps/maps.plugin.js : 313120 bytes
http://localhost:5620/9007199254740991/bundles/plugin/graph/graph.plugin.js : 317887 bytes
http://localhost:5620/9007199254740991/bundles/plugin/discover/discover.plugin.js : 410207 bytes
http://localhost:5620/9007199254740991/bundles/plugin/dashboard/dashboard.plugin.js : 571645 bytes
http://localhost:5620/9007199254740991/bundles/plugin/lens/lens.plugin.js : 652037 bytes
http://localhost:5620/9007199254740991/bundles/plugin/dashboardEnhanced/dashboardEnhanced.plugin.js : 183624 bytes
http://localhost:5620/9007199254740991/bundles/plugin/dashboardMode/dashboardMode.plugin.js : 573633 bytes
http://localhost:5620/9007199254740991/bundles/plugin/alerting/alerting.plugin.js : 28848 bytes
http://localhost:5620/9007199254740991/bundles/plugin/spaces/spaces.plugin.js : 347389 bytes
http://localhost:5620/9007199254740991/bundles/plugin/infra/infra.plugin.js : 619313 bytes
http://localhost:5620/9007199254740991/bundles/plugin/visualize/visualize.plugin.js : 288887 bytes
http://localhost:5620/9007199254740991/bundles/plugin/savedObjectsManagement/savedObjectsManagement.plugin.js : 299961 bytes
http://localhost:5620/9007199254740991/bundles/plugin/triggers_actions_ui/triggers_actions_ui.plugin.js : 325810 bytes
http://localhost:5620/9007199254740991/bundles/plugin/monitoring/monitoring.plugin.js : 292110 bytes
http://localhost:5620/9007199254740991/bundles/plugin/logstash/logstash.plugin.js : 306142 bytes
http://localhost:5620/9007199254740991/bundles/plugin/uptime/uptime.plugin.js : 307972 bytes
http://localhost:5620/9007199254740991/bundles/plugin/apm/apm.plugin.js : 358965 bytes
http://localhost:5620/9007199254740991/bundles/plugin/ml/ml.plugin.js : 739970 bytes
http://localhost:5620/node_modules/@kbn/ui-framework/dist/kui_light.css : 9847 bytes
http://localhost:5620/9007199254740991/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.css : 40018 bytes
http://localhost:5620/9007199254740991/bundles/commons.style.css : 7762 bytes
http://localhost:5620/9007199254740991/bundles/light_theme.style.css : 13759 bytes
http://localhost:5620/9007199254740991/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.v7.light.css : 64948 bytes
http://localhost:5620/9007199254740991/bundles/core/core.entry.js : 351361 bytes
http://localhost:5620/ui/fonts/inter_ui/Inter-UI-Regular.woff2 : 88428 bytes
http://localhost:5620/translations/en.json : 29 bytes
http://localhost:5620/api/core/capabilities : 784 bytes
http://localhost:5620/ui/fonts/inter_ui/Inter-UI-Medium.woff2 : 94368 bytes
http://localhost:5620/ui/fonts/inter_ui/Inter-UI-Light-BETA.woff2 : 96700 bytes
http://localhost:5620/ui/fonts/inter_ui/Inter-UI-SemiBold.woff2 : 94752 bytes
http://localhost:5620/app/home : 19947 bytes
http://localhost:5620/internal/spaces/_active_space : 134 bytes
http://localhost:5620/api/licensing/info : 366 bytes
http://localhost:5620/internal/security/me : 267 bytes
http://localhost:5620/internal/security/session : 119 bytes
http://localhost:5620/api/_newsfeed-FTS-external-service-simulators/kibana/v8.0.0.json : 453 bytes
http://localhost:5620/api/saved_objects/_bulk_get : 161 bytes
http://localhost:5620/ui/fonts/inter_ui/Inter-UI-Bold.woff2 : 94840 bytes
http://localhost:5620/api/saved_objects/_find?fields=title&per_page=1&search=*&search_fields=title&type=index-pattern : 52 bytes
http://localhost:5620/9007199254740991/built_assets/dlls/vendors_runtime.bundle.dll.js : 1738 bytes
http://localhost:5620/9007199254740991/built_assets/dlls/vendors_0.bundle.dll.js : 215296 bytes
http://localhost:5620/9007199254740991/built_assets/dlls/vendors_1.bundle.dll.js : 308854 bytes
http://localhost:5620/9007199254740991/built_assets/dlls/vendors_2.bundle.dll.js : 428005 bytes
http://localhost:5620/9007199254740991/built_assets/dlls/vendors_3.bundle.dll.js : 537777 bytes
http://localhost:5620/9007199254740991/bundles/commons.bundle.js : 706695 bytes
http://localhost:5620/9007199254740991/built_assets/dlls/vendors_0.style.dll.css : 31 bytes
http://localhost:5620/9007199254740991/built_assets/dlls/vendors_1.style.dll.css : 31 bytes
http://localhost:5620/9007199254740991/built_assets/dlls/vendors_2.style.dll.css : 31 bytes
http://localhost:5620/9007199254740991/built_assets/dlls/vendors_3.style.dll.css : 31 bytes
http://localhost:5620/9007199254740991/built_assets/css/plugins/timelion/index.light.css : 2925 bytes
http://localhost:5620/9007199254740991/built_assets/css/plugins/tests_bundle/index.light.css : 9851 bytes
http://localhost:5620/9007199254740991/built_assets/css/plugins/kibana/index.light.css : 22239 bytes
http://localhost:5620/9007199254740991/built_assets/css/plugins/maps/index.light.css : 7748 bytes
http://localhost:5620/9007199254740991/built_assets/css/core.light.css : 15416 bytes
http://localhost:5620/api/interpreter/fns : 1219 bytes
data:image/webp;base64,UklGRh4AAABXRUJQVlA4TBEAAAAvAQAAAAfQ//73v/+BiOh/AAA= : 0 bytes
http://localhost:5620/9007199254740991/bundles/1a11a1c1187b2a537ff50ca2e408ac0c.js : 94938 bytes
http://localhost:5620/api/timelion/functions : 6448 bytes
http://localhost:5620/api/uptime/index_status : 34 bytes
```

should something be filtered out, e.g. `http://localhost:5620/api` and `data:image` ?

Reply:
I think `/api/` and `/internal/`, partially because those are small but also because I don't think they'll have stable sizes. Anything that has a stable size will not actually be reported on PRs but only when a size change is detected will it be listed.

Reply:
At the moment, we are getting 97 assets after filtering out the ones that are not loaded not on every page, `data:`, `/api/`, `/internal/`

Reply:
Example report: https://github.com/spalger/kibana-ci-stats/blob/c430da57cfb597f8be043bae20958659b310752b/report.md#pr-66224